### PR TITLE
Add a PHP Composer (Packagist) finder

### PR DESCRIPTION
**Pull request by juledwar**
Adds the phpcompser finder type, which finds PHP packages at packagist.org.

- - -
This change is [Reviewable](https://reviewable.io/reviews/juledwar/soufi/62)

### CODE REVIEW

File: soufi/finders/php_composer.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        :return: A tuple of (URL, type) where the type is the archive_extension
+            to use.
+        """
+        url = f"{DEFAULT_INDEX}p2/{self.name}.json"
+        resp = requests.get(url, timeout=self.timeout)
+        if resp.status_code != requests.codes.ok:
+            raise exceptions.SourceNotFound
+
+        versions = resp.json()["packages"][self.name]
```
Comment (0xDEC0DE) [inline soufi/finders/php_composer.py:52]: This should catch/handle `KeyError`s, JSON parse errors, etc., if the response comes back malformed

File: soufi/finders/php_pecl.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def get_source_url(self):
 
 
 class PHPPECLDiscoveredSource(finder.DiscoveredSource):
-    """A discovered Rust Crate package."""
+    """A discovered PHP PECL package."""
```
Comment (0xDEC0DE) [inline soufi/finders/php_pecl.py:43]: LOL

File: soufi/testing/data/phpcomposer/monolog.json
Comment (0xDEC0DE) [inline soufi/testing/data/phpcomposer/monolog.json:1]: melting

**Review by 0xDEC0DE — CHANGES_REQUESTED** [review]

File: soufi/finders/php_pecl.py (hunk 2 above)
Comment (juledwar) [reply soufi/finders/php_pecl.py:43]: Copy pasta is the best kind of pasta.

**Review by 0xDEC0DE — APPROVED** [review]

**Pull request merged**